zetaclient/metrics/metrics_test.go

```diff
@@ -23,20 +28,52 @@ func (ms *MetricsSuite) SetUpSuite(c *C) {
 	ms.m = m
 }
 
+// assert that the curried metric actually uses the same underlying storage
+func (ms *MetricsSuite) TestCurryWith(c *C) {
+	rpcTotalsC := RPCCount.MustCurryWith(prometheus.Labels{"host": "test"})
+	rpcTotalsC.With(prometheus.Labels{"code": "400"}).Add(1.0)
+
+	rpcCtr := testutil.ToFloat64(RPCCount.With(prometheus.Labels{"host": "test", "code": "400"}))
+	c.Assert(rpcCtr, Equals, 1.0)
+
+	RPCCount.Reset()
+}
+
 func (ms *MetricsSuite) TestMetrics(c *C) {
 	GetFilterLogsPerChain.WithLabelValues("chain1").Inc()
 	GetFilterLogsPerChain.WithLabelValues("chain2").Inc()
 	GetFilterLogsPerChain.WithLabelValues("chain2").Inc()
 	time.Sleep(1 * time.Second)
-	res, err := http.Get("http://127.0.0.1:8886/metrics")
+
+	chain1Ctr := testutil.ToFloat64(GetFilterLogsPerChain.WithLabelValues("chain1"))
+	c.Assert(chain1Ctr, Equals, 1.0)
+
+	httpClient, err := GetInstrumentedHTTPClient("http://127.0.0.1:8886")
```

Comment:
Can you add some kind of api token here to check that it gets truncated in metrics labels?